utils/cooldownqueue.go

```diff
@@ -47,8 +48,17 @@ func makeEventKey(e watch.Event) string {
 	if !ok {
 		return ""
 	}
-	eventKey := string(object.GetUID())
-	return eventKey
+	id := domain.KindName{
+		Kind: &domain.Kind{
+			Group:    object.GetAPIVersion(),
+			Resource: object.GetKind(),
+		},
+		Name:            object.GetName(),
+		Namespace:       object.GetNamespace(),
+		ResourceVersion: domain.ToResourceVersion(object.GetResourceVersion()),
```

Comment:
@dwertent  `id.String()` takes into account only `Group`, `Version`, `Resource`, `Namespace`, `Name` so no need for the resource version...

Reply:
Also, the initialization of `domain.Kind` should be different...
```
&domain.Kind{
	Group:    res.Group,
	Version:  res.Version,
	Resource: res.Resource,
},
```